client/components/banner/test/index.jsx

```diff
@@ -1,165 +1,197 @@
-		expect( comp.find( Card ) ).toHaveLength( 0 );
-		expect( comp.find( 'DismissibleCard' ) ).toHaveLength( 1 );
+		renderWithRedux( <Banner { ...props } dismissPreferenceName={ 'banner-test' } /> );
+		expect( screen.getByLabelText( 'Dismiss' ) ).toBeInTheDocument();
 	} );
 
 	test( 'should have .has-call-to-action class if callToAction is defined', () => {
-		const comp = shallow( <Banner { ...props } callToAction={ 'Upgrade Now!' } /> );
-		expect( comp.find( '.has-call-to-action' ) ).toHaveLength( 1 );
+		const { container } = render( <Banner { ...props } callToAction={ 'Upgrade Now!' } /> );
+		expect( container.firstChild ).toHaveClass( 'has-call-to-action' );
 	} );
 
 	test( 'should not have .has-call-to-action class if callToAction is null', () => {
-		const comp = shallow( <Banner { ...props } callToAction={ null } /> );
-		expect( comp.find( '.has-call-to-action' ) ).toHaveLength( 0 );
+		const { container } = render( <Banner { ...props } callToAction={ null } /> );
+		expect( container.firstChild ).not.toHaveClass( 'has-call-to-action' );
 	} );
 
 	test( 'should render a <Button /> when callToAction is specified', () => {
-		const comp = shallow( <Banner { ...props } callToAction={ 'Buy something!' } /> );
-		expect( comp.find( Button ) ).toHaveLength( 1 );
+		render( <Banner { ...props } callToAction={ 'Buy something!' } /> );
+		expect( screen.getByRole( 'button' ) ).toHaveTextContent( 'Buy something!' );
 	} );
 
 	test( 'should not render a <Button /> when callToAction is not specified', () => {
-		const comp = shallow( <Banner { ...props } /> );
-		expect( comp.find( Button ) ).toHaveLength( 0 );
+		render( <Banner { ...props } /> );
+		expect( screen.queryByRole( 'button' ) ).toBeNull();
 	} );
 
 	test( 'should have .is-jetpack class and JetpackLogo if jetpack prop is defined', () => {
 		const { plan, ...propsWithoutPlan } = props;
-		const comp = shallow( <Banner { ...propsWithoutPlan } jetpack /> );
-		expect( comp.find( '.is-jetpack' ) ).toHaveLength( 1 );
-		expect( comp.find( 'JetpackLogo' ) ).toHaveLength( 1 );
+		const { container } = render( <Banner { ...propsWithoutPlan } jetpack /> );
+		expect( container.firstChild ).toHaveClass( 'is-jetpack' );
+		expect( container.getElementsByTagName( 'svg' ) ).toHaveLength( 1 );
 	} );
 
 	test( 'should render have .is-horizontal class if horizontal prop is defined', () => {
-		const comp = shallow( <Banner { ...props } horizontal /> );
-		expect( comp.find( '.is-horizontal' ) ).toHaveLength( 1 );
+		const { container } = render( <Banner { ...props } horizontal /> );
+		expect( container.getElementsByClassName( 'is-horizontal' ) ).toHaveLength( 1 );
 	} );
 
 	test( 'should render a <PlanPrice /> when price is specified', () => {
-		const comp = shallow( <Banner { ...props } price={ 100 } /> );
-		expect( comp.find( PlanPrice ) ).toHaveLength( 1 );
+		render( <Banner { ...props } price={ 100 } /> );
+		expect( screen.getByText( '100' ) ).toBeVisible();
```

Comment:
Hmm, how can we be sure that this is displayed in the price component and not just in the content of the banner?

Reply:
I was looking at this migration you did: https://github.com/Automattic/wp-calypso/blob/1d154291c9355b3eed50a96810210022bb29349b/client/components/language-picker/test/index.jsx#L50

Would that be a good approach for this one? 

`expect( screen.getByText( '100' ).parentNode ).toHaveClass( 'plan-price' );`

Reply:
That's a good example 👍 

The difference, though, is that the language slug appeared just in one place, so there wasn't any ambiguity when writing the test.

In this scenario, the problem is that we can have multiple price objects rendered there. We can have the original price and the discounted price, and we can't really tell if that text is the discounted price or the original one. We can also have a price range, and sometimes a tax amount. I guess, to be able to test this, we'd have to be testing the implementation details of `PlanPrice`, which is not recommended. 

So, how about we mock the `PlanPrice` component, and replace it with a simple component that renders that price only (coming from the prop we pass to the component)? Then, it would be easy enough to avoid testing implementation details, but we could directly test whether the passed `price` prop comes to the `PlanPrice` component as expected.